src/v/kafka/server/handlers/fetch.cc

```diff
@@ -702,6 +711,186 @@ class parallel_fetch_plan_executor final : public fetch_plan_executor::impl {
     }
 };
 
+class pid_controller {
```

Comment:
Class, method & member comments are really sparse or non-existent here. Not saying we need to add them now but it would be nice to see these more in future changes.